config/core/resources/pullsubscription.yaml

```diff
@@ -57,46 +64,57 @@ spec:
             secret:
               type: object
               description: "Credential used to poll the Cloud Pub/Sub Subscription. It is not used to create or delete the Subscription, only to poll it. The value of the secret entry must be a service account key in the JSON format (see https://cloud.google.com/iam/docs/creating-managing-service-account-keys). Defaults to secret.name of 'google-cloud-key' and secret.key of 'key.json'."
+              properties:
+                name:
+                  type: string
+                key:
+                  type: string
+                optional:
+                  type: boolean
             project:
               type: string
               description: "ID of the Google Cloud Project that the Pub/Sub Topic exists in. E.g. 'my-project-1234' rather than its display name, 'My Project' or its number '1234567890'. If omitted uses the Project ID from the GKE cluster metadata service."
             sink:
               type: object
               description: "Reference to an object that will resolve to a domain name to use as the sink."
-              anyOf:
-                - type: object
+              properties:
+                uri:
+                  type: string
+                  minLength: 1
+                ref:
+                  type: object
+                  required:
+                    - apiVersion
+                    - kind
+                    - name
                   properties:
-                    uri:
+                    apiVersion:
                       type: string
                       minLength: 1
-                - type: object
-                  properties:
-                    ref:
-                      type: object
-                      required:
-                        - apiVersion
-                        - kind
-                        - name
-                      properties:
-                        apiVersion:
-                          type: string
-                          minLength: 1
-                        kind:
-                          type: string
-                          minLength: 1
-                        name:
-                          type: string
-                          minLength: 1
+                    kind:
+                      type: string
+                      minLength: 1
+                    name:
+                      type: string
+                      minLength: 1
+              anyOf:
```

Comment:
I don't think we need this anyOf. It's not specified inhttps://github.com/knative/eventing/blob/master/config/core/resources/trigger.yaml

Reply:
This `anyOf` is for the `spec.sink` field, which is present on PullSubscriptions, but not Topics.

Reply:
Not sure why it matters that it's only on PullSubscription. `sink` is a Destination, similar to Trigger's `subscriber`. In particular, because sink has two properties and neither are required, I believe the semantics are already the same as what this anyOf describes.